Yosemite/YosemiteTests/Stores/ProductAttributeStoreTests.swift

```diff
@@ -0,0 +1,330 @@
+import XCTest
+@testable import Yosemite
+@testable import Storage
+@testable import Networking
+
+
+/// ProductAttributeStore Unit Tests
+///
+final class ProductAttributeStoreTests: XCTestCase {
+    /// Mock Network: Allows us to inject predefined responses!
+    ///
+    private var network: MockNetwork!
+
+    /// Mock Storage: InMemory
+    ///
+    private var storageManager: MockStorageManager!
+
+    /// Convenience Property: Returns the StorageType associated with the main thread.
+    ///
+    private var viewStorage: StorageType {
+        return storageManager.viewStorage
+    }
+
+    /// Convenience Property: Returns stored product attributes count.
+    ///
+    private var storedProductAttributesCount: Int {
+        return viewStorage.countObjects(ofType: Storage.ProductAttribute.self)
+    }
+
+    /// Store
+    ///
+    private var store: ProductAttributeStore!
+
+    /// Testing SiteID
+    ///
+    private let sampleSiteID: Int64 = 123
+
+    /// Testing Page Number
+    ///
+    private let defaultPageNumber = 1
+
+    // MARK: - Overridden Methods
+
+    override func setUp() {
+        super.setUp()
+        network = MockNetwork(useResponseQueue: true)
+        storageManager = MockStorageManager()
+        store = ProductAttributeStore(dispatcher: Dispatcher(),
+                                     storageManager: storageManager,
+                                     network: network)
+    }
+
+    override func tearDown() {
+        store = nil
+        network = nil
+        storageManager = nil
+
+        super.tearDown()
+    }
+
+    func test_synchronize_product_attributes_returns_attributes_upon_successful_response() throws {
+        // Given
+        network.simulateResponse(requestUrlSuffix: "products/attributes", filename: "product-attributes-all")
+        XCTAssertEqual(storedProductAttributesCount, 0)
+
+        // When
+        let result: Result<[Networking.ProductAttribute], Error> = try waitFor { promise in
```

Comment:
FYI: I don't think is necessary to explicitly set the result's type. But it may help ups with the compile times 🤔